### PR TITLE
Dockerize pep8speaks and use gunicorn to deploy

### PR DESCRIPTION
**Pull request by tushar5526**
Removed `heroku` conf files. 
Related to #192

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -8,3 +8,5 @@ unidiff~=0.6.0
 autopep8~=1.5.2
 markdown~=3.2.2
 beautifulsoup4~=4.9.1
+markupsafe==2.0.1
```
Comment (Mr-Sunglasses) [inline requirements/base.txt]: we should add a version constraint to it ie. `markupsafe~=2.0.1`

File: requirements/base.txt [hunk 2]
```diff
@@ -8,3 +8,5 @@ unidiff~=0.6.0
...
+python-dotenv==1.0.0
```
Comment (Mr-Sunglasses) [inline requirements/base.txt]: we should add a version constraint to it ie. `python-dotenv~=1.0.0
`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Use an official Python runtime as a parent image
```
Comment (Mr-Sunglasses) [inline Dockerfile:1]: Can we try a multistage build to reduce the image size, as we are running it on a low-spec machine?

```
# Use an official Python runtime as a parent image and Build environment
FROM python:3.8-slim AS build

# Set the working directory to /app
WORKDIR /app

# Copy the current directory contents into the container at /app
COPY . /app

# Install any needed packages specified in requirements.txt
RUN pip install --no-cache-dir -r requirements.txt

# Stage 2: Production environment
FROM python:3.8-slim AS production

# Set the working directory to /app
WORKDIR /app

# Copy only the necessary files from the build stage to the production image
COPY --from=build /app /app

# Expose port 8000 for the Gunicorn server to listen on
EXPOSE 8000

# Define the command to run your application using Gunicorn
CMD ["gunicorn", "server:app", "--bind", "0.0.0.0:8000", "--workers", "4"]

```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+version: '3'
+services:
+  pep8speaks:
+    build: .
+    image: pep8speaks
```
Comment (Mr-Sunglasses) [inline docker-compose.yml:5]: tag it to some version semantics eg. `pep8speaks:latest`

**Review by Mr-Sunglasses — CHANGES_REQUESTED** [review]
I think these changes should me made, everything else LGTM.
btw awesome work @tushar5526 👍🏻

File: Dockerfile (hunk 3 above)
Comment (tushar5526) [reply Dockerfile:1]: I will check if multistage will make much difference here, mostly multistage is vital when you need to install other "apt" packages to install the packages. I should add the `--no-cache-dir` though.

Comment (tushar5526) [reply Dockerfile:1]: Switched to alpine, using a multi-stage build is helpful when you install other packages to install the main dependencies.

File: docker-compose.yml (hunk 4 above)
Comment (tushar5526) [reply docker-compose.yml:5]: This does not matter in local builds, planning to switch to GitHub packages all together soon, to avoid building images.

**Review by Mr-Sunglasses — APPROVED** [review]
LGTM

**Pull request merged**